//
// Copyright 2018 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package asylo;

import "asylo/enclave.proto";

// This file contains extensions to the enclave proto messages for the whole-
// application wrapper.

// Represents command-line arguments.
message CommandLineArgs {
  // The command-line arguments, in order.
  repeated string arguments = 1;
}

extend EnclaveConfig {
  // The command-line arguments to be passed to the application.
  optional CommandLineArgs command_line_args = 215413541;
}

extend EnclaveOutput {
  // The return value from the application's main() function.
  optional int64 main_return_value = 215413541;
}
